src/main/kotlin/org/rust/cargo/commands/Cargo.kt

```diff
@@ -53,6 +55,7 @@ class Cargo(
             .withParameters(command)
             .withParameters(additionalArguments)
             .withEnvironment(environmentVariables)
+            .withEnvironment(CargoConstants.RUSTC_ENV_VAR, pathToRustExecutable)
```

Comment:
I think this should come before `withEnvironment(environmentVariables)` to be able to override `RUSTC`.